torch/fx/experimental/symbolic_shapes.py

```diff
@@ -258,16 +300,21 @@ def magic_impl(self, other):
         other_expr = self.shape_env.replace(other_expr)
         out = func(expr, other_expr)
         out = sympy.expand(out)
-        if method in ["truediv"]:
+        if method in always_float_magic_methods:
             return PySymFloat(out, self.shape_env)
+        elif method in always_int_magic_methods:
+            return PySymInt(out, self.shape_env)
         else:
             # TODO: relational operators actually technically return a
             # PySymBool, this is a type error
-            return py_type(out, self.shape_env)
+            if py_type is PySymFloat or isinstance(other, PySymFloat):
```

Comment:
Both our PySymInt and PySymFloat might contain sympy bool! Our custom __bool__ will make sure this goes smothly

Reply:
hmm, what does this line actually have to do with sympy bool? I thought that this line is just the "type promotion" line - where for most math ops, mixed int/float compute should give you a float.